src/box/lua/connpool.lua

```diff
@@ -271,46 +316,102 @@ local function filter(opts)
             table.insert(static_candidates, instance_name)
         end
     end
-    -- Return if retrieving dynamic information is not required.
-    if next(static_candidates) == nil or next(dynamic_opts) == nil then
+
+    if mode == 'any' then
         return static_candidates
     end
 
-    -- Filter the remaining candidates after connecting to them.
-    local connected_candidates = connect_to_candidates(static_candidates)
+    local connected_instances = acquire_all_candidates(static_candidates)
+
```

Comment:
This comment also seems useful: it marks the point where the actual network activity is performed. Helpful when you're not deeply in context of this code.